### ISSUE TITLE
Translations in other languages

### ISSUE THREAD

**Issue by nlehuby** [OPENER-next/OpenStop#148]
### Idea

Would you consider to make the app available in other languages ?

OpenStreetMap is an international project and collect accessibility data in other countries could be great too. I've tried to use the app to gather data around my home but did not succeed as I don't speak German :sad

### Implementation

The translation itself can be done by the community (here, or with third-party tools services as Transifex or POEditor).

On the other hand, it is necessary to :
* extract and publish the list of displayed labels and texts, so that they can be translated
* in the app: have a way to choose a language for the display / detect the phone language and switch automatically to it

**Comment by 7h30n3** [comment]
Hi @nlehuby 
thank you for your feature request. I'm glad to see that OpenStop is used in other countries. For that reason we also have thought about an internationalization of the app. With our code base it should be fairly easy to be implemented. However it would still be a time consuming task.

We try to work on a solution or hope that we can find some spare time to implement this feature.
We are also always happy if we get pull requests :)

**Comment by Suburbanno** [comment]
I would also like to help with the translation to start using where I live!

**Comment by habi** [comment]
[Every Door](https://every-door.app) is translated at Weblate](https://hosted.weblate.org/projects/every-door/app/), which offers free plans for libre projects: https://hosted.weblate.org/hosting/

**Comment by habi** [comment]
[Vespucci](http://vespucci.io/tutorials/faq/) uses Transifex, which [is free for OSS](https://www.transifex.com/open-source/).

**Comment by matheusgomesms** [comment]
I would also love to translate this app! Every Door (OSS), as mentioned, is also built in Flutter and has translations enabled on Weblate. As a first step, maybe we can translate the .arb files?

Maybe these help:
https://github.com/Zverik/every_door/issues/101
https://github.com/Zverik/every_door/commit/44a96934687dcacd46d19e685c9bb767f38b610d

**Comment by chrisdebian** [comment]
Is there a way to ask users if they can help? For example, a mailing list, or a message through the app?

Does the app gather different data to StreetComplete?

Thanks,

Chris | chris_debian

**Comment by mnalis** [comment]
> Is there a way to ask users if they can help? For example, a mailing list, or a message through the app?

Ummm, ask ***which*** users (OpenStop users, or OSM users generally?)
And asking them to help ***about what*** ?  About using the the OpenStop app to add missing data? About making PR for code so OpenStop supports translations? About translating OpenStop to different languages (obviously, only possible once a code to support it is there)?

> Does the app gather different data to StreetComplete?

In short, yes (at least currently).

You can look at https://github.com/OPENER-next/OpenStop/blob/master/assets/datasets/question_catalog.json - even if you don't understand questions in German, if you search for `osm_tags` under `answers`, it will give you the idea.

While some quests seem about the same (like `fee=yes|no` on `amenity=parking`) others are not present in StreetComplete, and vice versa (e.g. _"Hat die Fahrplan-Anzeige für die gesamte Station eine akustische Ausgabe?"_ / _"Does the schedule display for the entire station have an audible output?"_ to tag `departures_board:speech_output=yes|no` is present in OpenStop and not present in StreetComplete).

So while StreetComplete is more generic and has more quests about everything else, OpenStop is (as noted in its [README.md](https://github.com/OPENER-next/OpenStop/blob/master/README.md)) specifically oriented more towards _accessibility data for public transport_, and is more detailed in that segment.


Also, it seems that OpenStop works on both Android and iOS devices too, while StreetComplete is [currently limited](https://github.com/streetcomplete/StreetComplete/issues/1892) to Android devices only.

**Comment by chrisdebian** [comment]
The topic is about translations, and the users are the users of this app.

**Comment by mnalis** [comment]
> The topic is about translations, and the users are the users of this app.

Yeah, I know it is about translations. But if you read comments above, there are several different parts that needs to be done before app can be shown in your language:
- general multi-language support needs to be programmed for all strings in OpenStop, and all strings moved to that format
- custom UI (detect system language, set default, allow user selecting language, etc.) needs to be programmed
- translation infrastructure needs to be setup (Transifex / POeditor / Weblate / ...) and integrated in build workflow
- somebody needs to actually translate from German to other languages.

First three are the ones than need to be completed before fourth can even begin. And if I understand correctly, that is what the volunteer programmers are [currently being asked for help with](https://github.com/OPENER-next/OpenStop/issues/148#issuecomment-1456070814).

So, I was unsure if you were talking about asking users somewhere for that **_programming help_** _now_ (i.e. first three tasks which are needed at this point), **or** if you were talking longer-term  about asking OpenStop users for **_help for translating into their language_** (i.e. fourth task) at some indefinite time in the future (but hopefully sooner than never!) 

Because if the latter, no OpenStop (nor other) users can start translating before there is something they can translate (i.e. before first three tasks are already implemented), and (I guess) most OpenStop users are unlikely to be able to help with first three tasks, where help is currently wanted.

**Comment by chrisdebian** [comment]
Sorry :-(

**Comment by Robbendebiene** [comment]
@mnalis You perfectly summarized the current state and what needs to be done. Thanks for answering the questions.

**Short update from our side:** We will probably get further support in our team soon. The main task of our new supporter will be to implement the necessary foundation (steps 1 to 3) in order to enable others to translate the app.

So @chrisdebian and others, stay tuned 🙂

**Comment by Mannivu** [comment]
I was looking of there was a way to translate this app and I've found this thread. Not to pressure anyone, but there are other platforms that can be used (but I don't know anything about prices, I'm a translator, not a dev, sorry), e.g. [Crowdin](https://crowdin.com/) (used by Mastodon and UBlock Origin among others) and some other projects don't use an external platform, but they require the user to download an app, like [Joplin](https://github.com/laurent22/joplin#localisation).

**Comment by mnalis** [comment]
And I can translate to Croatian once English (and translating infrastructure) is available.

**Comment by 7h30n3** [comment]
We have finally released OpenStop with multi-language-support. It should soon be available on Google Play, App Store and F-Droid or you can download the apk directly from GitHub if you can't wait any longer.
For now it's only German and English and some elements aren't fully translated yet, so this issue stays open for a bit longer.
There will also be an integration of a web translation tool, but we haven't decided yet which one we use, but https://weblate.org/ looks promising.

**Comment by LaoshuBaby** [comment]
First of all, I would like to thank the developers for their continued support, but it seems that there are still more strings that default to German. In addition, I‘ve tested in the just updated 0.5.0 and the language cannot be manually specified in the settings interface?

<img src="https://github.com/OPENER-next/OpenStop/assets/42690037/d53e0141-2764-4646-84d3-77b734a59927" width=250px/>

---

(BTW, Reply to above about every_door, because this APP  is interactive question based and only focuses on the public transportation field, so I think it is very useful for simplifying the work process and better promoting it to local public transportation enthusiasts who have no experience in OSM, and helps to better maintain data. This is its feature and why I expect a  localized languages support.)

**Comment by Robbendebiene** [comment]
@LaoshuBaby Thanks for the report. The app UI is actually translated but we missed explicitly defining English as the default language (and due to alphabetical sorting it picked German). The discrepancy occurs because we have two slightly different translation engines, one for questions and one for the app UI.

As a workaround for now you can add English as a supported language to your device. It should then display the UI in English.

This report also helped us to fix another problem where questions were not translated in the correct language when multiple languages where defined. Please keep testing, but keep in mind that markers and most importantly OSM changesets are not translated yet.

> I‘ve tested in the just updated 0.5.0 and the language cannot be manually specified in the settings interface?

We will probably add this at a certain point. For now we focus on full translation support for the next release which is almost done 🙂

**Comment by angoca** [comment]
What is the way to translate into another language? Some months ago, I have already prepared the translation to Spanish, and I am ready to include that here: https://github.com/MaptimeBogota/Varios/tree/main/OpenStop-app

**Comment by Robbendebiene** [comment]
@angoca Cool! We are planning to use a tool like [Weblate](https://weblate.org/).
In the meantime you could create a new "arb" file here to translate the questions and answers:

- https://github.com/OPENER-next/OpenStop/tree/master/assets/question_catalog/locales
- https://github.com/OPENER-next/OpenStop/tree/master/assets/advanced_question_catalog/locales

For translating the UI a new "arb" file needs to be created here:
- https://github.com/OPENER-next/OpenStop/tree/master/lib/l10n

**Comment by Robbendebiene** [comment]
Just wanted to give an update here.

With [version 0.6.0](https://github.com/OPENER-next/OpenStop/releases/tag/v0.6.0) the translation into english is completed.
This means that theoretically the app can now be translated into different languages by manually editing the `.arb` files.
However to make translating easier and more convenient we are working on integrating a translation tool. See #221

Thank you for your support.

**Comment by 7h30n3** [comment]
As mentioned in #221 a web tool is now available to translate OpenStop in different languages.
